src/interpreter.rs

```diff
@@ -18,6 +64,99 @@ impl Execute for Expression {
     }
 }
 
+#[derive(Debug, Clone, Copy)]
+pub enum OperandItem<'a> {
+    Val(bool),
+    Predicate(&'a Predicate),
+}
+fn evaluate_operand_item(item: OperandItem, ctx: &mut Context, m: &mut Match) -> bool {
+    match item {
+        OperandItem::Val(b) => b,
+        OperandItem::Predicate(p) => p.execute(ctx, m),
+    }
+}
+
+impl Execute for Route {
+    fn execute(&self, ctx: &mut Context, m: &mut Match) -> bool {
+        //operand stack
+        let mut top: usize = 0;
+        let mut operand_stack: [OperandItem; 2] = [OperandItem::Val(false); 2];
+
+        for item in &self.stack {
+            match item {
+                RouteTerm::LogicalOperator(op) => {
+                    match op {
+                        RouteLogicalOperators::And => {
+                            //stack pop
+                            top -= 1;
+                            let left = evaluate_operand_item(operand_stack[top], ctx, m);
```

Comment:
Seems the we use postorder iteration to store expressions in `self.stack` so the stack would like: `[left, right, op, left, right, op]`. and the operand stack will also be [left, right] when we iterating `self.stack` here. so when we "pop" the operand stack, seems we will first get "right" then "left"? That will not affect the correctness, but maybe make accurate naming will make it easy to understand.

Reply:
Correct. Fixed. thanks.